### ISSUE TITLE
Optionally including username to password promp

### ISSUE THREAD

**Issue by delicb** [dbcli/pgcli#823]
This is suggestion for improvement, not a bug.

I am using `pgcli` quite often. My terminal app is iTerm2 which has builtin password manager. It recognizes regex in terminal and, when triggered, offers to enter password read from KeyChain. Quite handy feature.

Problem is that, with `pgcli`, there is no single regex to use for different databases, since prompt is "Password:". I would like to make distinction based on user I am using to connect.

Currently I patch local installation of `pgcli` to include username (which solves my problem, but more generic solution would probably include both username and database, maybe even host) in password prompt. 

These are two places that I have modified:
https://github.com/dbcli/pgcli/blob/master/pgcli/main.py#L368
https://github.com/dbcli/pgcli/blob/master/pgcli/main.py#L386

My modification looks something like this:
```python
passwd = click.prompt('%s password' % user, hide_input=True,
                       show_default=False, type=str)
```

I understand if you do not want to change default behavior, but would you consider adding flag that would modify password prompt to include username (and maybe database and host)? If yes, I might be able to provide PR.

Thanks.

**Comment by j-bennet** [comment]
Certainly! This is a valid use case. If you wanted to change the default behavior to ask for `Password (<username>):` (or something along those lines) instead of `Password:`, I don't think it would be a problem either. It's a one-time prompt that quickly goes away, and even if someone does not like extra verbosity, it's not staying on screen long enough to be annoying.

**Comment by amjith** [comment]
@delicb If you have a local patch that you're currently using why not make it a PR. If we have modifications or suggestions we can discuss it in the PR. It is easier (faster) to get a feature in if there is a PR. 😄

**Comment by delicb** [comment]
@j-bennet Great, thank you. I will go with changing default behavior then, since it seem as most straight forward way, no unnecessary complications. 

@amjith Certainly, I will prepare PR promptly. 

Thank you both for being so cooperative :).

**Comment by j-bennet** [comment]
Included in 1.8.2 (just released).
